crypto/crypto/src/x25519.rs

```diff
@@ -1,340 +1,151 @@
-    type DHSharedKeyMaterial = X25519SharedKey;
+//
+// Underlying Implementation
+// =========================
+//
+// We re-export the dalek-x25519 library,
+// This makes it easier to uniformalize build dalek-x25519 in libra-core.
+//
 
-    // Diffie-Hellman exchange
-    fn dh(self, their_public: &X25519PublicKey) -> X25519SharedKey {
-        let shared_secret = self.0.diffie_hellman(&their_public.0);
-        X25519SharedKey(shared_secret)
-    }
-}
+pub use x25519_dalek;
 
-//////////////////////
-// X25519StaticPrivateKey Traits //
-//////////////////////
+//
+// Main types and constants
+// ========================
+//
 
-impl X25519StaticPrivateKey {
-    /// Derives a keypair `(X25519PrivateKey, X25519PublicKey)` from
-    /// a) salt (optional) - denoted as 'salt' in RFC 5869
-    /// b) seed - denoted as 'IKM' in RFC 5869
-    /// c) application info (optional) - denoted as 'info' in RFC 5869
-    ///
-    /// using the HKDF key derivation protocol, as defined in RFC 5869.
-    /// This implementation uses the full extract-then-expand HKDF steps
-    /// based on the SHA-256 hash function.
-    pub fn derive_keypair_from_seed(
-        salt: Option<&[u8]>,
-        seed: &[u8],
-        app_info: Option<&[u8]>,
-    ) -> (X25519StaticPrivateKey, X25519StaticPublicKey) {
-        let derived_bytes =
-            Hkdf::<Sha256>::extract_then_expand(salt, seed, app_info, X25519_PRIVATE_KEY_LENGTH);
-        let mut key_bytes = [0u8; X25519_PRIVATE_KEY_LENGTH];
-        key_bytes.copy_from_slice(derived_bytes.unwrap().as_slice());
+/// Size of a X25519 private key
+const PRIVATE_KEY_SIZE: usize = 32;
+/// Size of a X25519 public key
+const PUBLIC_KEY_SIZE: usize = 32;
 
-        let secret: x25519_dalek::StaticSecret = x25519_dalek::StaticSecret::from(key_bytes);
-        let public: x25519_dalek::PublicKey = (&secret).into();
-        (
-            X25519StaticPrivateKey(secret),
-            X25519StaticPublicKey(X25519PublicKey(public)),
-        )
-    }
+/// This type should be used to deserialize a received private key
+#[derive(DeserializeKey, SilentDisplay, SilentDebug, SerializeKey)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary, Clone, PartialEq))]
+pub struct PrivateKey([u8; PRIVATE_KEY_SIZE]);
 
-    /// Generates a random keypair `(PrivateKey, PublicKey)` by combining the output of `EntropyRng`
-    /// with a user-provided seed. This concatenated seed is used as the seed to HKDF (RFC 5869).
-    ///
-    /// Similarly to `derive_keypair_from_seed` the user provides the following inputs:
-    /// a) salt (optional) - denoted as 'salt' in RFC 5869
-    /// b) seed - denoted as 'IKM' in RFC 5869
-    /// c) application info (optional) - denoted as 'info' in RFC 5869
-    ///
-    /// Note that this method is not deterministic, but the (random + static seed) key
-    /// generation makes it safer against low entropy pools and weak RNGs.
-    pub fn generate_keypair_hybrid(
```

Comment:
You are removing an important security feature here. Hybrid key generation (secret seed + randomness) is considered to combine the pros from both just RNG and deterministic generation).

Reply:
this isn't used anywhere though